From 8f0140ace401197beaed1df6a48f259756d8bbd9 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Tue, 8 Aug 2017 19:06:16 +0300
Subject: [PATCH] tests: robustify run-strip-nothing.sh against unstripped
 libc_nonshared.a

When glibc's libc_nonshared.a contains objects with debug info,
this debug info is leaked into every output file produced by gcc.
Change run-strip-nothing.sh to use "gcc -s" instead of plain "gcc"
for producing objects without debug info.

Signed-off-by: Dmitry V. Levin <ldv@altlinux.org>
---
 tests/ChangeLog            | 4 ++++
 tests/run-strip-nothing.sh | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/ChangeLog b/tests/ChangeLog
index 04efdc8..94f4cf2 100644
--- a/tests/ChangeLog
+++ b/tests/ChangeLog
@@ -1,3 +1,7 @@
+2017-08-08  Dmitry V. Levin <ldv@altlinux.org>
+
+	* run-strip-nothing.sh: Add -s.
+
 2017-07-26  Mark Wielaard  <mark@klomp.org>
 
 	* dwarf-getmacros.c (mac): Use DW_MACRO names instead of DW_MACRO_GNU.
diff --git a/tests/run-strip-nothing.sh b/tests/run-strip-nothing.sh
index e80bd90..914fdfb 100755
--- a/tests/run-strip-nothing.sh
+++ b/tests/run-strip-nothing.sh
@@ -23,7 +23,7 @@
 tempfiles a.out strip.out debug.out
 
 # Create no-debug a.out.
-echo "int main() { return 1; }" | gcc -xc -
+echo "int main() { return 1; }" | gcc -s -xc -
 
 # strip to file
 testrun ${abs_top_builddir}/src/strip -g -o strip.out ||
-- 
2.9.3

